src/Development/Rattle/Shared.hs

```diff
@@ -43,7 +42,12 @@ getList :: (Show a, Serialize b) => String -> Shared -> a -> IO [b]
 getList typ (Shared lock dir) name = withLock lock $ do
     let file = dir </> typ </> filename (hashString $ show name)
     b <- doesFileExist file
-    if not b then return [] else fromRight [] . decode <$> BS.readFile file
+    if not b then return [] else decodeAll <$> BS.readFile file
+      where decodeAll bstr
+              | BS.null bstr = []
+              | otherwise = case runGetState get bstr 0 of
+                              Left str -> error $ "Failed to decode: " ++ str
+                              Right (a, rbstr) -> decodeAll rbstr ++ a
```

Comment:
This will be O(n^2) with the appends. I guess we should return the list the way it is in the file, and then if someone wants the end, make them do last? Having the list in chronological order seems to make most intuitive sense.